Setting_up_Travis_CI_and_Coveralls.md

```diff
@@ -0,0 +1,107 @@
+# Setting up Travis-CI and Coveralls for a KBase Github repo
+This document will walk through the steps involved in activating Travis-CI for continuous, automated testing, along with Coveralls for reporting code coverage of tests.
+
+[Background and automated testing](#background)  
+[Travis-CI initial setup](#travis-init)  
+[Travis-CI script configuration](#travis-config)  
+[Coveralls initial setup](#coveralls-init)  
+[Coveralls configuration](#coveralls-config)  
+[KBase Challenges](#kbase-challenges)  
+
+## <a name="background"></a>Background and automated testing
+
+Using Github gives us numerous tools to help with automatic unit testing of open source projects. Travis-CI (https://travis-ci.org) automatically downloads and tests your repo, returning whether or not all tests pass. If you tests output a .coverage file (see [below](#coveralls-config)), that can be automatically passed to Coveralls (https://coveralls.io/) where the amount of code tested can be reported in your repo.
```

Comment:
I'd make the links here regular MD urls like `[Travis-CI](https://travis-ci.org)`, that way the rendered doc feels even more 'html-like' :)